nixos/modules/services/hardware/hddfancontrol.nix

```diff
@@ -0,0 +1,64 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.hddfancontrol;
+  types = lib.types;
+in
+
+{
+  options = {
+
+    services.hddfancontrol.enable = lib.mkEnableOption "hddfancontrol daemon";
+
+    services.hddfancontrol.disks = lib.mkOption {
+      type = types.listOf types.str;
+      default = [];
+      description = ''
+        Drive(s) to get temperature from
+      '';
+      example = ["/dev/sda"];
+    };
+
+    services.hddfancontrol.pwm_paths = lib.mkOption {
+      type = types.listOf types.str;
+      default = [];
+      description = ''
+        PWM filepath(s) to control fan speed (under /sys)
+      '';
+      example = ["/sys/class/hwmon/hwmon2/pwm1"];
+    };
+
+    services.hddfancontrol.use_smartctl = lib.mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Probe temperature using smartctl instead of hddtemp or hdparm
+      '';
+    };
+
+    services.hddfancontrol.extra_args = lib.mkOption {
+      type = types.str;
+      default = "";
+      description = ''
+        Extra commandline arguments for hddfancontrol
+      '';
+      example = "--pwm-start-value 32 --pwm-stop-value 0 --spin-down-time 900";
+    };
+  };
```

Comment:
I think it would be useful if you were allowed to specify these as a list. That seems to be [the most common way to make `extraArgs`](https://search.nixos.org/options?channel=23.05&from=0&size=50&sort=relevance&type=packages&query=extraArgs), although it varies a bit